### PR TITLE
Described the diff between Quarkiverse and a Quarkus platform

### PR DESCRIPTION
**Pull request by aloubyansky**

### CODE REVIEW

**Comment by loicmathieu** [comment]
The comparison with the platform give the idea that a Quarkiverse extension has no sort of testing with Quarkus.

Maybe add some notes of the fact that Quarkiverse extensions are tested nighlty on the Quarkus master branch but that the garantee of compability to a new release is on the extension author responsability.

This again raise my question of "compatibility matrix" and how we can enforce it, document it, ... and the tooling needed to it.

For exemple, on Quarkus 1.10 a changes has been made to quarkus-grpc, so the next version of my Google Cloud Services will depends on Quarkus 1.10.

But if a user adds the Quarkus 1.9 BOM + my extension on versiion 0.3 (the future version that will be compatible with Quarkus 1.10 only) Maven will resolves the Quarkus dependencies on 1.9 as the BOM will takes priority to my extension dependencies ... This is the result for not being part of the Platform (not uniform dependency resolution).

**Comment by aloubyansky** [comment]
> The comparison with the platform give the idea that a Quarkiverse extension has no sort of testing with Quarkus.

The paragraph above though does mention `Automated Cross-testing of your extension with Quarkus builds/releases`. I wasn't sure whether I wanted to repeat that in this paragraph, especially when, imo, the primary difference between the two is not about Quarkus release testing. The key difference here is the compatibility between the extensions. Platform extensions are compatible among themselves while the Quarkiverse ones (in general) might not be, no such promise. The platform release process implies coordination between the participants, while the Quarkiverse members aren't obliged to, they are independent.

> Maybe add some notes of the fact that Quarkiverse extensions are tested nighlty on the Quarkus master branch but that the garantee of compability to a new release is on the extension author responsability.
> 
> This again raise my question of "compatibility matrix" and how we can enforce it, document it, ... and the tooling needed to it.

Right. That remains an open question. We don't have a great answer to that atm except it should be documented and the users should consult the docs. This is not the end of the story, of course. For now we are just trying to set the basic things up and get the extensions listed. We'll be addressing the compatibility next. We have some prototypes that will help with this.

> For exemple, on Quarkus 1.10 a changes has been made to quarkus-grpc, so the next version of my Google Cloud Services will depends on Quarkus 1.10.
> 
> But if a user adds the Quarkus 1.9 BOM + my extension on versiion 0.3 (the future version that will be compatible with Quarkus 1.10 only) Maven will resolves the Quarkus dependencies on 1.9 as the BOM will takes priority to my extension dependencies ... This is the result for not being part of the Platform (not uniform dependency resolution).

Right, so the next version of your extension becomes incompatible with Quarkus 1.9.x. That should and will be properly handled by the tools.

File: docs/Home.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ for some reason cannot do a release we can in the best scenario help or in the w
...
 
+# Quarkiverse and Quarkus Platform
+
+The word Quarkiverse, derived from "Quarkus" and "universe", is carrying the meaning of the universe of Quarkus extensions. In other words, it is intended to represent a part of the Quarkus ecosystem. However, at the same time, there is also a notion of a Quarkus platform. There is a significant difference between the two.
+
+While the extensions under the Quarkiverse organization are generally independent projects lead by independent development teams, the fundamental promise of a Quarkus platform is any combination of the Quarkus extensions the platform consists of can be used in the same application without causing any conflict for each other. Quarkus platform implies coordination, cross testing and dependency version alignment across extensions that constitue the platform which is not generally applicable to the Quarkiverse extensions. It is, however, still possible for extensions hosted in the Quarkiverse to join a Quarkus platform assuming they satisfy the purpose and policies defined for that platform.
```
Comment (gastaldi) [inline docs/Home.md]:
```suggestion
While the extensions under the Quarkiverse organization are generally independent projects lead by independent development teams, the fundamental promise of a Quarkus platform is any combination of the Quarkus extensions the platform consists of can be used in the same application without causing any conflict for each other. Quarkus platform implies coordination, cross testing and dependency version alignment across extensions that constitute the platform which is not generally applicable to the Quarkiverse extensions. It is, however, still possible for extensions hosted in the Quarkiverse to join a Quarkus platform assuming they satisfy the purpose and policies defined for that platform.
```

**Comment by gastaldi** [comment]
I'd like to have that merged soon. Any objections?

**Pull request merged**